Apps/FCM/ios/FCMSampleApp/AppDelegate.swift

```diff
@@ -0,0 +1,131 @@
+    
+    // Set FCM messaging delegate
+    Messaging.messaging().delegate = self
+    
+    // Uncomment only if you want to register for remote push when the app starts
+    // UIApplication.shared.registerForRemoteNotifications()
+    
+    return true
+  }
+  
+  // Deep links handling for app scheme links
+  func application(_ app: UIApplication, open url: URL, options: [UIApplication.OpenURLOptionsKey : Any] = [:]) -> Bool {
+    return RCTLinkingManager.application(app, open: url, options: options)
+  }
+  
+  // Deep links handling for universal links
+  func application(_ application: UIApplication, continue userActivity: NSUserActivity, restorationHandler: @escaping ([UIUserActivityRestoring]?) -> Void) -> Bool {
+    guard let url = userActivity.webpageURL else {
+      return false
+    }
+    
+    let universalLinkUrl = URL(string: "http://www.amiapp-reactnative-fcm.com")
+
+    // Return true from this function if your app handled the link.
+    // Return false from this function if app does not handle the link and you want sdk to open the URL in a browser.
+    if (url.scheme == "http" || url.scheme == "https") && url.host == universalLinkUrl?.host {
+      return RCTLinkingManager.application(application, continue: userActivity, restorationHandler: restorationHandler)
+    } else {
+      return false
+    }
+  }
+}
+
+extension AppDelegate: RCTBridgeDelegate {
+  func sourceURL(for bridge: RCTBridge!) -> URL! {
+    return RCTBundleURLProvider.sharedSettings().jsBundleURL(forBundleRoot: "index")
+  }
+}
+
+extension AppDelegate: MessagingDelegate {
+  func messaging(_ messaging: Messaging, didReceiveRegistrationToken fcmToken: String?) {
+    MessagingPush.shared.messaging(messaging, didReceiveRegistrationToken: fcmToken)
+  }
+  
+  func application(_ application: UIApplication, didFailToRegisterForRemoteNotificationsWithError error: Error) {
+      MessagingPush.shared.application(application, didFailToRegisterForRemoteNotificationsWithError: error)
+  }
+}
+
+extension AppDelegate: UNUserNotificationCenterDelegate {
+    // OPTIONAL: If you want your push UI to show even with the app in the foreground, override this function and call
+    // the completion handler.
+    @available(iOS 10.0, *)
+    func userNotificationCenter(
+        _ center: UNUserNotificationCenter,
+        willPresent notification: UNNotification,
+        withCompletionHandler completionHandler: @escaping (UNNotificationPresentationOptions)
+            -> Void
+    ) {
+        completionHandler([.list, .banner, .badge, .sound])
+    }
+
+    // Function that gets called when push notification clicked
+    func userNotificationCenter(
+        _ center: UNUserNotificationCenter,
+        didReceive response: UNNotificationResponse,
+        withCompletionHandler completionHandler: @escaping () -> Void
+    ) {
+        // Send Customer.io SDK click event to process. This enables features such as
```

Comment:
I would suggest to use the [javascript methods](https://customer.io/docs/sdk/react-native/push-notifications/push-metrics/#metrics-javascript) instead of this native code. We are moving towards removing the dependency from native modules and handling functionality using javascript methods.

Reply:
I don't think the javascript method is exactly equivalent for this because:

- As per my understanding, our recommendation is still to implement `userNotificationCenter` in iOS, I don't know if we changed this? Our [docs also still suggests this implementation](https://www.customer.io/docs/sdk/react-native/push-notifications/push/?codelang=FCM%2FSwift#)
- We do not have any other notification library so there is no right place from where we can call equivalent javascript methods
- The JS method linked only tracks metrics and not handles deep links

Reply:
Ah ! Right. My bad. 
We still depend on other libraries to receive payload. 👍